lib/Reference/SearchablePageReferenceProvider.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace OCA\Collectives\Reference;
+
+use DateTime;
+use OC\Collaboration\Reference\LinkReferenceProvider;
+use OCA\Collectives\Model\CollectiveInfo;
+use OCA\Collectives\Service\CollectiveService;
+use OCA\Collectives\Service\PageService;
+use OCP\Collaboration\Reference\ADiscoverableReferenceProvider;
+use OCP\Collaboration\Reference\ISearchableReferenceProvider;
+use OCP\Collaboration\Reference\Reference;
+use OC\Collaboration\Reference\ReferenceManager;
+use OCA\Collectives\AppInfo\Application;
+use OCP\Collaboration\Reference\IReference;
+use OCP\Files\File;
+use OCP\IDateTimeFormatter;
+use OCP\IL10N;
+
+use OCP\IURLGenerator;
+
+class SearchablePageReferenceProvider extends ADiscoverableReferenceProvider implements ISearchableReferenceProvider {
```

Comment:
* `PageReferenceProvider` can only be used with Nextcloud 25. It only includes the "reference widget" stuff (to render link previews)
* `SearchablePageReferenceProvider` can be used with Nextcloud >= 26. It contains `PageReferenceProvider` plus the link picker stuff:
    * discoverable means it will be listed as a provider in the link picker
    * searchable means it has to declare a list of supported unified search providers (that will be used by the link picker once "Collectives" is selected)